core/src/main/java/org/elasticsearch/ResourceNotFoundException.java

```diff
@@ -16,31 +16,37 @@
  * specific language governing permissions and limitations
  * under the License.
  */
-
-package org.elasticsearch.indices;
+package org.elasticsearch;
 
 import org.elasticsearch.common.io.stream.StreamInput;
-import org.elasticsearch.index.Index;
-import org.elasticsearch.index.IndexException;
 import org.elasticsearch.rest.RestStatus;
 
 import java.io.IOException;
 
 /**
- *
+ * Generic ResourceNotFoundException corresponding to the {@link RestStatus#NOT_FOUND} status code
  */
-public class IndexMissingException extends IndexException {
+public class ResourceNotFoundException extends ElasticsearchException {
+
+    public ResourceNotFoundException(String msg, Object... args) {
+        super(msg, args);
+    }
 
-    public IndexMissingException(Index index) {
-        super(index, "no such index");
+    protected ResourceNotFoundException(String msg, Throwable cause, Object... args) {
+        super(msg, cause, args);
     }
 
-    public IndexMissingException(StreamInput in) throws IOException{
+    public ResourceNotFoundException(StreamInput in) throws IOException {
         super(in);
     }
 
     @Override
-    public RestStatus status() {
+    public final RestStatus status() {
         return RestStatus.NOT_FOUND;
     }
-}
\ No newline at end of file
+
+    @Override
+    public void setResources(String type, String... id) {
```

Comment:
do we need to have this method? can we just make the parent one public, given that we have a public getResources there as well?